pkg/prometheus/promcfg.go

```diff
@@ -1802,5 +1816,9 @@ func (l *limitEnforcer) addLimitsToYAML(cfg yaml.MapSlice, k limitKey, limit uin
 		return cfg
 	}
 
-	return append(cfg, yaml.MapItem{Key: k.prometheusField, Value: getLimit(limit, enforcedLimit)})
+	// Since BodySizeLimit is defined only in PrometheusCRD
+	if k.specField != "enforcedBodySizeLimit" {
```

Comment:
do we need a special case? Since `limit` is always 0 for `enforcedBodySizeLimit`, I would assume that `getLimit(limit, enforcedLimit)` would return `enforcedLimit` if it's > 0. Anyway it would make sense to have the comment on the caller side explaining why we pass a zero value.